source/Planet.h

```diff
@@ -64,11 +65,17 @@ class Planet {
 	// Get planet's noun descriptor from attributes
 	const std::string &Noun() const;
 	
-	// Check whether there is a spaceport (which implies there is also trading,
-	// jobs, banking, and hiring).
-	bool HasSpaceport() const;
-	// Get the spaceport's descriptive text.
-	const std::string &SpaceportDescription() const;
+	// Check whether there is a port (which may even be a full spaceport).
+	bool HasPort() const;
+	// Get this planet's port. Might be empty if there is no port.
+	const Port &GetPort() const;
+	// Check whether there are port services (such as trading, jobs, banking, and hiring)
+	// available on this planet.
+	bool HasServices() const;
+	// Check whether the given recharging is possible on this planet.
+	bool CanRecharge(int type) const;
+	// Check whether the given service is available on this planet.
+	bool IsAvailable(int type) const;
```

Comment:
I think I'd prefer `HasService` as the function name so that it's more clear what we're checking is available.